### ISSUE TITLE
Path to my spec files insn't complete

### ISSUE THREAD

**Issue by sodiac** [vitest-dev/vscode#179]
When i have in my `src` directory some directories they named like as the followed one's
- src/core
- src/collections
- src/conditionals

Is the name inside of my Vitest explorer like
- re/result/result.spec.ts
- llections/dictionary/dictionary.spec.ts
- nditionals/match/match.spec.ts

My fist asumetion is that every directory is starting with the characters `co`.
If ranemed all directories like

- src/sore
- src/sollections
- src/sonditionals

or

- src/sire
- src/sillections
- src/sinditionals

This has the same effect, that the path isn't complete, inside of the Vitest explorer

**To Reproduce**

1. Rename all directories inside of your src folder with the same two characters. `co, si, fo`
2. Click on Vitest to see the path to your spec files
2.1. Vitest view settings `Sort by Location` and `View as Tree`

**Expected behavior**
I can see the full path to my spec files

**Screenshots**
<img width="960" alt="grafik" src="https://github.com/vitest-dev/vscode/assets/3404305/5904665d-d803-42bc-b5e8-f179e23e56f5">

**Environment**

info.txt
```json
{
  "System": {
    "OS": "macOS 13.5.1"
  },
  "Binaries": {
    "Node": {
      "version": "16.17.1",
      "path": "~/.nvm/versions/node/v16.17.1/bin/node"
    },
    "Yarn": "Not Found",
    "npm": {
      "version": "8.15.0",
      "path": "~/.nvm/versions/node/v16.17.1/bin/npm"
    }
  },
  "IDEs": {
    "VSCode": {
      "version": "1.81.1",
      "path": "/usr/local/bin/code"
    }
  },
  "npmPackages": {
    "vite": {
      "installed": "2.9.16",
      "wanted": "^2.9.9"
    },
    "vitest": {
      "installed": "0.12.10",
      "wanted": "^0.12.6"
    }
  }
}
```

vite.config.ts
```typescript
{
  "System": {
    "OS": "macOS 13.5.1"
  },
  "Binaries": {
    "Node": {
      "version": "16.17.1",
      "path": "~/.nvm/versions/node/v16.17.1/bin/node"
    },
    "Yarn": "Not Found",
    "npm": {
      "version": "8.15.0",
      "path": "~/.nvm/versions/node/v16.17.1/bin/npm"
    }
  },
  "IDEs": {
    "VSCode": {
      "version": "1.81.1",
      "path": "/usr/local/bin/code"
    }
  },
  "npmPackages": {
    "vite": {
      "installed": "2.9.16",
      "wanted": "^2.9.9"
    },
    "vitest": {
      "installed": "0.12.10",
      "wanted": "^0.12.6"
    }
  }
}
```

vitest.config.ts
```typescript
import tsconfigPaths from 'vite-tsconfig-paths';
import { defineConfig } from 'vitest/config';

export default defineConfig({
	plugins: [tsconfigPaths()],
	test: {
		globals: true,
		exclude: ['**/node_modules/**'],
		include: ['**/*.spec.ts'],
		coverage: {
			provider: 'istanbul',
			reporter: ['html'],
			reportsDirectory: './.tests/coverage',
		},
		reporters: ['default', 'html'],
		outputFile: './.tests/index.html',
	},
});
```

tsconfig.json
```json
{
	"compilerOptions": {
		"target": "ES2020",
		"useDefineForClassFields": true,
		"module": "ESNext",
		"lib": ["ES2020"],
		"skipLibCheck": true,
		"baseUrl": "./src",
		"paths": {
			"core": ["./core/."],
			"conditional": ["./conditionals/."],
			"collections": ["./collections/."]
		},

		/* Bundler mode */
		"moduleResolution": "bundler",
		"allowImportingTsExtensions": true,
		"resolveJsonModule": true,
		"isolatedModules": true,
		"noEmit": true,

		/* Linting */
		"strict": true,
		"noUnusedLocals": false,
		"noUnusedParameters": false,
		"noFallthroughCasesInSwitch": true
	},
	"include": ["src"]
}
```

package.json
```json
{
	....
	"type": "module",
	"files": [
		"dist"
	],
	"main": "./dist/std.umd.cjs",
	"module": "./dist/std.js",
	"types": "./dist/index.d.ts",
	"engines": {
		"npm": ">=8.15.0 <=9.6.7",
		"node": ">=16.17.1 <=18.17.1"
	},
	"exports": {
		".": {
			"import": "./dist/std.js",
			"require": "./dist/std.umd.cjs"
		}
	},
	"scripts": {
		"build": "tsc && vite build",
		"serve:tests": "vitest --coverage --ui",
		"lint:check": "eslint src --ext .ts --ext .tsx && tsc --noEmit",
		"postinstall": "husky install"
	},
	"devDependencies": {
		"@types/node": "20.5.9",
		"@typescript-eslint/eslint-plugin": "6.5.0",
		"@typescript-eslint/parser": "6.5.0",
		"@vitest/coverage-istanbul": "0.34.3",
		"@vitest/ui": "0.34.3",
		"eslint": "8.48.0",
		"eslint-plugin-check-file": "2.6.2",
		"eslint-plugin-no-null": "1.0.2",
		"husky": "8.0.3",
		"prettier": "3.0.3",
		"typescript": "5.2.2",
		"vite": "4.4.9",
		"vite-plugin-dts": "3.5.3",
		"vite-tsconfig-paths": "4.2.0",
		"vitest": "0.34.3"
	}
}

```

- OS: macOS Ventura 13.5.1
- VSCode version: 1.81.1 (Universal)
- Node.js version: 16.17.1
- Vitest version: 0.34.3
- Vitest plugin version: v0.2.42

**Comment by sodiac** [comment]
**UPDATE**
Even when i update node to the latest version, still exists this issue

**Comment by huwshimi** [comment]
It appears that the test explorer truncates file names where the path matches for all test files, even if that is part way through a folder name.

For example for this folder structure:

<img width="228" alt="Screenshot 2023-10-18 at 4 37 59 pm" src="https://github.com/vitest-dev/vscode/assets/361637/ca0170b2-885d-4c3d-bfe2-0f4c4a50d6c1">

It truncates everything up to the end of `../outer`:

<img width="245" alt="Screenshot 2023-10-18 at 4 37 44 pm" src="https://github.com/vitest-dev/vscode/assets/361637/b19b6cfe-ca18-45c4-8653-1d00cdc01979">

So if we were to add another test file at the top level:

<img width="227" alt="Screenshot 2023-10-18 at 4 48 53 pm" src="https://github.com/vitest-dev/vscode/assets/361637/361d32af-bde2-4b38-be18-13a121c7d971">

Then less of the path would be the same between all test files, so more gets displayed:

<img width="279" alt="Screenshot 2023-10-18 at 4 49 32 pm" src="https://github.com/vitest-dev/vscode/assets/361637/889dfabc-09b7-45bf-8e53-c1b3ea8d55bf">

I think this is just the behaviour of the test explorer, and as far as I can tell there isn't a way to configure this to behave differently.

**Comment by sheremet-va** [comment]
This should be fixed in pre-release 0.5.0 and higher. Note that the extension now requires Vitest 1.4.0 or higher.
